reference/test/stop/combined.cpp

```diff
@@ -124,8 +137,7 @@ TEST_F(Combined, WaitsTillTime)
     auto start = std::chrono::steady_clock::now();
 
     for (int i = 0; i < testiters; i++) {
-        std::this_thread::sleep_for(
-            std::chrono::duration<double>(timelimit / testiters));
+        sleep_millisecond(1);
```

Comment:
Please don't hard-code numbers when it was previously dependent on a `constexpr` variable. If the time is not enough, just change variable `timelimit` to an integer (here, I guess `10`), and keep the calculation in the call `sleep_millisecond`.
Also: Can windows not sleep for a shorter period of time? This test in its current form is much weaker than it was previously (by factor 1000).
I would actually prefer if the windows test had different numbers if it can't handle the linux ones, and the Linux test staying the same (with a comment of course stating that the resolution in Windows is much lower than the one on Linux).

Reply:
I do not think the test is weaker than before.
The solver set `time_limit` and `max_loop`.
We choose `run_loop` which is less than `max_loop`.
Sleep `time_limit`/`run_loop` for `run_loop` should exceed `time_limit` to stop the solver when Sleep guarantees sleeping for more than the given time.

Reply:
I just looked at it again and apparently, we only test if: `time_d >= 1e-9 - 1e-4`, so you are right, this test was very lax to say the least.
You are right, this test is better now than it was before, my bad for not reading it in more detail.

Reply:
I would still prefer these numbers to be a `constexpr` variable, as we had before. That way, we can adapt the behavior more easily, without the need to change multiple numbers which are in correlation.